python/turbodbc_test/test_select_arrow.py

```diff
@@ -0,0 +1,247 @@
+from collections import OrderedDict
+from mock import patch
+
+import datetime
+import pytest
+import turbodbc
+
+from query_fixture import query_fixture
+from helpers import open_cursor, for_each_database, for_one_database, generate_microseconds_with_precision
+
+# Skip all parquet tests if we can't import pyarrow.parquet
+pa = pytest.importorskip('pyarrow')
+
+# Ignore these with pytest ... -m 'not parquet'
+pyarrow = pytest.mark.pyarrow
+
+
+def _fix_case(configuration, string):
+    """
+    some databases return column names in upper case
+    """
+    capabilities = configuration['capabilities']
+    if capabilities['reports_column_names_as_upper_case']:
+        return string.upper()
+    else:
+        return string
+
+
+@for_one_database
+@pyarrow
+def test_no_arrow_support(dsn, configuration):
+    with open_cursor(configuration) as cursor:
+        cursor.execute("SELECT 42")
+        with patch('turbodbc.cursor._has_arrow_support', return_value=False):
+            with pytest.raises(turbodbc.Error):
+                cursor.fetchallarrow()
+
+
+@for_one_database
+@pyarrow
+def test_arrow_without_result_set_raises(dsn, configuration):
+    with open_cursor(configuration) as cursor:
+        with pytest.raises(turbodbc.InterfaceError):
+            cursor.fetchallarrow()
+
+
+@for_each_database
+@pyarrow
+def test_arrow_empty_column(dsn, configuration):
+    with open_cursor(configuration) as cursor:
+        with query_fixture(cursor, configuration, 'INSERT INTEGER') as table_name:
+            cursor.execute("SELECT a FROM {}".format(table_name))
+            result = cursor.fetchallarrow()
+            assert isinstance(result, pa.Table)
+            assert result.num_columns == 1
+            assert result.num_rows == 0
+
+
+@for_each_database
+@pyarrow
+def test_arrow_int_column(dsn, configuration):
```

Comment:
These tests are _very_ similar to each other. Try to extract a common test function.